tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SQS/AwsSqsCommon.cs

```diff
@@ -56,5 +58,31 @@ public static Scope CreateScope(Tracer tracer, string operation, out AwsSqsTags
             // or we couldn't populate it completely (some tags is better than no tags)
             return scope;
         }
+
+        public static string GetQueueName(string queueUrl)
+        {
+            if (queueUrl is null)
+            {
+                return queueUrl;
+            }
+
+            var lastSeparationIndex = queueUrl.LastIndexOf('/') + 1;
```

Comment:
Just a small note I'm leaving for later 😆:

Instead of doing a substring operation each time, we might consider using a ConditionalWeakTable to map the queueUrl to a previously computed result. This would be good to microbenchmark and prove that there's a tangible benefit to the new approach.

I've been using a ConditionalWeakTable everywhere in these peer.service PR's because they seem to operate well as a cache that _does not_ grow infinitely. If the original object (in this case the queueUrl string) is no longer referenced and is garbage-collected, then the corresponding cached result will also be garbage-collected.